main.js

```diff
@@ -1,163 +1,352 @@
+
+
+// 1.a
+const tvName = inventory.map((tv) => tv.name);
+console.log(tvName);
+
+
+// 1.b
+function soldOut(list) {
+    return list.filter((sold) => sold.originalStock === sold.sold);
+}
+
+console.log(soldOut(inventory));
+
+
+// 1.c
+function hasAmbiLight(list) {
+    return list.filter((inventory) => inventory.options.ambiLight);
+}
+
+console.log(hasAmbiLight(inventory));
+
+
+// 1.c
+function lowestToHighestPrice(list) {
+    return list.sort((a, b) => a.price - b.price);
+}
+
+
+// Opdracht 2.
+
+
+// 2.a
+let totalSum = 0;
+
+for (let i = 0; i < inventory.length; i++) {
+    totalSum += inventory[i].sold;
+}
+
+
+// 2.b
+const soldTvs = document.getElementById("sold-tv's");
+
+const soldTvsTitle = document.createElement("h3");
+soldTvsTitle.textContent = "Aantal verkochte tv's"
+
+const allSoldTvs = document.createElement("h4");
+allSoldTvs.textContent = totalSum;
+
+soldTvs.appendChild(soldTvsTitle);
+soldTvs.appendChild(allSoldTvs);
+
+
+// 2.c
+let totalOriginalStock = 0;
+
+for (let i = 0; i < inventory.length; i++) {
+    totalOriginalStock += inventory[i].originalStock;
+}
+
+
+// 2.d
+console.log(totalOriginalStock);
+
+const originalStock = document.getElementById("sum-of-tv's-bought");
+
+const originalStockTitle = document.createElement("h3");
+originalStockTitle.textContent = "aantal tv's ingekocht";
+
+const allOriginalStock = document.createElement("h4");
+allOriginalStock.textContent = totalOriginalStock;
+
+originalStock.appendChild(originalStockTitle);
+originalStock.appendChild(allOriginalStock);
+
+
+// 2.e
+const amountLeftToSell = totalOriginalStock - totalSum;
+
+const amountOfTVsToSell = document.getElementById("tv's-left-to-sale");
+
+const amountLeftToSellTitle = document.createElement("h3");
+amountLeftToSellTitle.textContent = "aantal tv's nog te verkopen";
+
+const allAmountLeftToSell = document.createElement("h4");
+allAmountLeftToSell.textContent = amountLeftToSell;
+
+amountOfTVsToSell.appendChild(amountLeftToSellTitle);
+amountOfTVsToSell.appendChild(allAmountLeftToSell);
+
+
+// Opdracht 3.
+
+function printList(list) {
+    const megaList = document.getElementById("tv-brand-names");
+
+    list.map((listTv) => {
+        const newList = document.createElement("li");
+        newList.textContent = listTv.brand;
+
+        megaList.appendChild(newList);
+    })
+}
+
+printList(inventory);
+
+
+// Opdracht 4.
+
+
+// 4.a
+function formatTvName(tv) {
+    return `${tv.brand} ${tv.type} - ${tv.name}`;
+}
+
+// 4.b
+function priceTv(tv) {
+    return `€${tv.price},-`;
+}
+
+// 4.c
+function sizeTv(tv) {
+    let sizes = "";
+    for (let i = 0; i < tv.availableSizes.length; i++) {
+        if (i < tv.availableSizes.length - 1) {
+            sizes = sizes + `${tv.availableSizes[i]} inch (${Math.floor(tv.availableSizes[i] * 2.54)} cm) | `;
+        } else {
+            sizes = sizes + `${tv.availableSizes[i]} inch (${Math.floor(tv.availableSizes[i] * 2.54)} cm)`;
+        }
+    }
+    return sizes;
```

Comment:
Goed gedaan, maar ik heb een tip hoe het nog korter kan 😉 
```
return tv.availableSizes.map(size => `${size} inch (${Math.floor(size * 2.54)} cm)`).join(" | ");
```

Reply:
dat is inderdaad een stuk korter :)